src/Generics/SOP/Constraint.hs

```diff
@@ -42,6 +42,24 @@ import Generics.SOP.Sing
 --
 -- means that 'f' can assume that all elements of the n-ary
 -- product satisfy 'Eq'.
+
+-- Note on superclasses: ghc cannot deduce superclasses from 'All'
+-- constraints.
+-- You might expect the following to compile
+-- > class (Eq a) => MyClass a
```

Comment:
there must be an empty line (with `--` though) to split the paragraph from code block